src/main/java/org/folio/coursereserves/util/CRUtil.java

```diff
@@ -515,202 +416,153 @@ public static Future<JsonObject> lookupItemByBarcode(String barcode,
-            logger.info("PermanentLocationObject lookup failed {}",
-                permLocationFuture.cause().getMessage());
+          if (processingStatusFuture.succeeded()) {
+            ProcessingStatusObject pso = new ProcessingStatusObject();
+            copyFields(pso, processingStatusFuture.result());
+            reserve.setProcessingStatusObject(pso);
           }
-        } else {
-          logger.info("No copied item field in reserve to populate");
-        }
-        if(processingStatusFuture.succeeded()) {
-          ProcessingStatusObject pso = new ProcessingStatusObject();
-          copyFields(pso, processingStatusFuture.result());
-          reserve.setProcessingStatusObject(pso);
-        }
-        if(copyrightStatusFuture.succeeded()) {
-          CopyrightStatusObject cso = new CopyrightStatusObject();
-          copyFields(cso, copyrightStatusFuture.result());
-          reserve.getCopyrightTracking().setCopyrightStatusObject(cso);
-        }
-        if(loanTypeFuture.succeeded()) {
-          TemporaryLoanTypeObject tlto = temporaryLoanTypeObjectFromJson(
-              loanTypeFuture.result());
-          reserve.setTemporaryLoanTypeObject(tlto);
-        }
-        promise.complete();
-      } catch(Exception e) {
-        promise.fail(e);
-      }
-    });
-    return promise.future();
+          if (copyrightStatusFuture.succeeded()) {
+            CopyrightStatusObject cso = new CopyrightStatusObject();
+            copyFields(cso, copyrightStatusFuture.result());
+            reserve.getCopyrightTracking().setCopyrightStatusObject(cso);
+          }
+          if (loanTypeFuture.succeeded()) {
+            TemporaryLoanTypeObject tlto = temporaryLoanTypeObjectFromJson(
+                loanTypeFuture.result());
+            reserve.setTemporaryLoanTypeObject(tlto);
+          }
+          return null;
+        });
   }
 
   public static Future<CourseListing> lookupExpandedCourseListing(String courseListingId,
       Map<String, String> okapiHeaders, Context context, Boolean expandTerm) {
-    Promise<CourseListing> promise = Promise.promise();
-    getCourseListingById(courseListingId, okapiHeaders, context).onComplete(clRes -> {
-      if(clRes.failed()) {
-        promise.fail(clRes.cause());
-      } else if(clRes.result() == null) {
-        promise.complete(null);
+    return getCourseListingById(courseListingId, okapiHeaders, context).compose(courselisting -> {
+      if (courselisting == null) {
+        return Future.succeededFuture(null);
+      }
+      String termId = courselisting.getTermId();
+      String courseTypeId = courselisting.getCourseTypeId();
+      String locationId = courselisting.getLocationId();
+      String servicepointId = courselisting.getServicepointId();
+      Future<Term> termFuture;
+      Future<CourseType> coursetypeFuture;
+      Future<JsonObject> locationFuture;
+      Future<JsonObject> servicePointFuture;
+      if (expandTerm && termId != null) {
```

Comment:
All these multiple `if (expandTerm` can be avoided by using
```
    if (! expandTerm) {
      return Future.succeededFuture(courselisting);
    }
```
once.

Reply:
OK. But we can do even better. expandTerm is always true. I'll remove the parameter and reduce complexity that way.